cmd/rpcdaemon/commands/trace_filtering.go

```diff
@@ -23,19 +23,73 @@ import (
 )
 
 // Transaction implements trace_transaction
-// TODO(tjayrush): I think this should return an []interface{}, so we can return both Parity and Geth traces
 func (api *TraceAPIImpl) Transaction(ctx context.Context, txHash common.Hash) (ParityTraces, error) {
 	tx, err := api.kv.BeginRo(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer tx.Rollback()
 
-	traces, err := api.getTransactionTraces(tx, ctx, txHash)
+	txn, _, blockNumber, _ := rawdb.ReadTransaction(ethdb.NewRoTxDb(tx), txHash)
```

Comment:
I think you should also keep the 4th return value (`txIndex`) and use it later to only execute until that `txIndex` later

Reply:
Done